.github/workflows/docs-preview-prepare.yml

```diff
@@ -33,11 +28,21 @@ jobs:
     runs-on: ubuntu-24.04
     steps:
       - uses: actions/checkout@v4
+        with:
+          # As `on: pull_request_target` runs this workflow from the default branch of our repo,
+          # Adjust to checkout the correct PR branch (including repo to support PRs from forks):
+          ref: ${{ env.PR_REF }}
+          repository: ${{ env.PR_REPO }}
+          # Prevent `secrets.GITHUB_TOKEN` from being stored in a `.git/config` file, which could otherwise
+          # be compromised when executing untrusted code from a PR (as is done in the build step below).
+          persist-credentials: false
 
       # ================== #
       # Build docs preview #
       # ================== #
 
+      # CAUTION: Security risk due to executing `build-docs.sh` (which a PR could modify to be malicious):
+      # Care must be taken to run this workflow as if it were in an untrusted context (like the `on: pull_request` event trigger would).
```

Comment:
The only reason we allow this is so that an update to the Docker image tag via PR would trigger the workflow to detect any breakage to docs generation.

I could make another `workflow_call` workflow file for this instead (_[`docs-production-deploy.yml` requires calling `build-docs.sh`](https://github.com/docker-mailserver/docker-mailserver/blob/6b4627ceab44c65d04ae25926aab05956ca2bafb/.github/workflows/docs-production-deploy.yml#L45) too_), and perhaps with `uses:` excluding the repo and branch pin, we'd still be able to verify that version bump PR works correctly 🤔